graphql/src/main/java/com/linecorp/armeria/common/scalar/PathScalar.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.scalar;
+
+import java.nio.file.Path;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import graphql.schema.Coercing;
+import graphql.schema.CoercingParseLiteralException;
+import graphql.schema.CoercingParseValueException;
+import graphql.schema.CoercingSerializeException;
+import graphql.schema.GraphQLScalarType;
+
+/**
+ * A path scalar that accepts file stings and produces {@link Path} objects at runtime.
```

Comment:
Maybe simpler this way? Let me know if it's not correct:

```suggestion
 * A path scalar that converts a file path string into a {@link Path}.
```